lambda/app/src/controllers/requests.ts

```diff
@@ -200,6 +219,170 @@ export const createRequest = async (session: Session, data: IntegrationData) =>
+    bcscClientId = clientResponse.data.client_id;
+  }
+
+  const idpCreated = await getIdp(env, integration.clientId);
+  if (!idpCreated) {
+    await createIdp(
+      {
+        alias: integration.clientId,
+        displayName: `BC Services Card - ${integration.clientId}`,
+        enabled: true,
+        storeToken: true,
+        providerId: 'oidc',
+        realm: 'standard',
+        config: {
+          clientId: bcscClientId,
+          clientSecret: bcscClientSecret,
+          authorizationUrl: `${bcscBaseUrl}/login/oidc/authorize`,
+          tokenUrl: `${bcscBaseUrl}/oauth2/token`,
+          userInfoUrl: `${bcscBaseUrl}/oauth2/userinfo`,
+          jwksUrl: `${bcscBaseUrl}/oauth2/jwk`,
+          syncMode: 'IMPORT',
+          disableUserInfo: true,
+          clientAuthMethod: 'client_secret_post',
+          validateSignature: true,
+          useJwksUrl: true,
+          defaultScope: bcscDefaultScopes,
+        },
+      },
+      env,
+    );
+  }
+
+  const idpMappers = await getIdpMappers({
+    environment: env,
+    idpAlias: integration.clientId,
+  });
+
+  const createIdpMapperPromises = bcscIdpMappers.map((mapper) => {
+    const alreadyExists = idpMappers.some((existingMapper) => existingMapper.name === mapper.name);
+    if (!alreadyExists) {
+      return createIdpMapper({
+        environment: env,
+        name: mapper.name,
+        idpAlias: integration.clientId,
+        idpMapper: mapper.type,
+        idpMapperConfig: {
+          claim: mapper.name,
+          attribute: mapper.name,
+          syncMode: 'FORCE',
+          template: mapper.template,
+        },
+      });
+    }
+  });
+
+  await Promise.all(createIdpMapperPromises);
+
+  const clientScopeData = {
+    environment: env,
+    realmName: 'standard',
+    scopeName: integration.clientId,
+  };
+
+  let clientScope = await getClientScope(clientScopeData);
+  if (!clientScope) {
+    clientScope = await createClientScope(clientScopeData);
+  }
+
+  await getClientScopeMapper({
+    environment: env,
+    scopeId: clientScope.id,
+    mapperName: 'attributes',
+  }).then((mapperExists) => {
+    if (!mapperExists) {
+      createClientScopeMapper({
+        environment: env,
+        realmName: 'standard',
+        scopeName: clientScope.name,
+        protocol: 'openid-connect',
+        protocolMapper: 'oidc-idp-userinfo-mapper',
+        protocolMapperName: 'attributes',
+        protocolMapperConfig: {
+          signatureExpected: true,
+          'user.attribute': 'attributes',
+          userAttributes: integration.bcscAttributes?.join(','),
+          decodeUserInfoResponse: true,
+          'claim.name': 'attributes',
+          'jsonType.label': 'String',
+          'id.token.claim': true,
+          'access.token.claim': true,
+          'userinfo.token.claim': true,
+        },
+      });
+    }
+  });
+};
+
+export const deleteBCSCIntegration = async (request: BCSCClientParameters, keycoakClientId: string) => {
```

Comment:
@jlangy lets not delete the BCSC registrations but we can introduce additional column called status and upon deletion, we can simply archive it